docs/changehistory/NextVersion.md

```diff
@@ -334,6 +347,80 @@ the widget in a floating container. The property `defaultFloatingPosition` may a
 
 The method `getFloatingWidgetContainerIds()` has been added to FrontstageDef to retrieve the Ids for all floating widget containers for the active frontstage as specified by the `frontstageDef`. These ids can be used to query the size of the floating container via `frontstageDef.getFloatingWidgetContainerBounds`. The method `frontstageDef.setFloatingWidgetContainerBounds` can then be used to set the size and position of a floating widget container.
 
+#### `ControlledTree` API Changes
+
+There were some breaking API changes related to `ControlledTree` component:
+
+* The component now takes `TreeModel` rather than `VisibleTreeNodes` as a prop to avoid requiring consumers to manage `VisibleTreeNodes` object. As a result, the `useVisibleTreeNodes` hook was replaced with `useTreeModel` hook. Typical mitigation:
+
+  **Before:**
+
+  ```tsx
+  const visibleNodes = useVisibleTreeNodes(modelSource);
+  return (
+    <ControlledTree {...otherProps} visibleNodes={visibleNodes} />
+  );
+  ```
+
+  **After:**
+
+  ```tsx
+  const treeModel = useTreeModel(modelSource);
+  return (
+    <ControlledTree {...otherProps} model={treeModel} />
+  );
+  ```
+
+* Name of the `treeEvents` prop was changed to `eventsHandler` to make it clearer. Typical mitigation:
+
+  **Before:**
+
+  ```tsx
+  return (
+    <ControlledTree {...otherProps} treeEvents={eventsHandler} />
+  );
+  ```
+
+  **After:**
+
+  ```tsx
+  return (
+    <ControlledTree {...otherProps} eventsHandler={eventsHandler} />
+  );
+  ```
+
+* Made the props `width` and `height` required. Previously they were optional and forced us to use non-optimal approach when not provided. Now it's up to the consumer to tell the size of the component. Typical mitigation:
+
+  **Before:**
+
+  ```tsx
+  return (
+    <ControlledTree {...props} />
+  );
+  ```
+
+  **After:**
+
+  ```tsx
+  const width = 100;
+  const height = 100;
+  return (
+    <ControlledTree {...props} width={width} height={height} />
+  );
+  ```
+
+  Or, the size may be determined dynamically. The below example uses an `AutoSizer` component from `react-virtualized-auto-sizer` package for that:
```

Comment:
We should direct people to use [`ResizeObserver`](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) instead.

Reply:
I don't mind us referencing `ResizeObserver` as a possible solution, but I don't want to give an example with it, because it would be much larger. What do you think about rephrasing the comment like:
```markdown
Or, the size may be determined dynamically using [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) or some third-party library. The below example uses an `AutoSizer` component from `react-virtualized-auto-sizer` package for that:
```
?

Reply:
Changed to
```markdown
`width` and `height` props may be calculated dynamically using [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) API.
```
and removed the example - let's leave it up to consumers to decide how to come up with width and height as it's fairly straightforward to find and anything we suggest may become out of date quickly.

Reply:
Agree, thanks.